### PR TITLE
Support a "Clipping Rectangle" in anchored positioning behavior

### LINKED ISSUE #1129: Allow Anchored Positioning behavior to work in a non-full-height container element
**Describe the bug**
One of the requirements of the Anchored Positioning behavior is that the nearest _positioned_ ancestor of the floating element needs to be of sufficient height to fit the entire floating element. However, many UI's rely on the fact that this container is normally rendered with `overflow: visible` (or `scroll` in some cases). We should account for this when determining the acceptable area to draw a floating element.

**To Reproduce**
1. Use the `anchoredPosition` behavior to pushing a floating element when its container is `document.body`.

**Expected behavior**
If `document.body` has no styles applied that alter its height, and the contents of the page are otherwise small, the floating element will get positioned in an unexpected spot.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by T-Hugs**
This change updates the anchored positioning behavior so that the floating element now respects its "clipping" container. This is the first ancestor element of the floating element that does _not_ have its `overflow` style set to anything other than `visible`. Furthermore, the clipping container must either be _positioned_ or be an ancestor of a _positioned_ ancestor of the floating element.

With this in place, we can now special-case `document.body` to take the full height of the window. This allows us to avoid having to manually create an element with full height just to contain the floating element.

**In other words, we now support positioning the floating element outside the bounds of a container as long as it will render visibly.**

Closes #1129 

### Screenshots
Screenshot of the new story that demonstrates the change:
image

### Merge checklist
- [x] Added or updated TypeScript definitions (`index.d.ts`) if necessary
- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [ ] Tested in Edge

### CODE REVIEW

**Review by keithamus — APPROVED** [review]
Diff is interesting to see, it becomes evident we're mostly changing calculations from one containing element to another.

This LGTM :+1:

**Review by keithamus — APPROVED** [review]
Nice tests :+1:
